### PR TITLE
Changes for NVIDIA HPC SDK 24.5

### PR DESCRIPTION
**Pull request by jmuddnv**
Add HPC SDK 24.5 to etc/config/c++.amazon.properties
Add CUDA 12.4 to etc/config/cuda.amazon.properties

### CODE REVIEW

**Comment by jmuddnv** [comment]
HPC SDK 24.5 supports CUDA 12.4, which I added to etc/config/cuda.amazon.properties, but apparently CUDA 12.4 has not been added to compiler-explorer.  I don't know how to do that, and I'm not on the CUDA team.  Can someone help me navigate how to resolve this issue?

**Comment by partouf** [comment]
related https://github.com/compiler-explorer/infra/pull/1314

File: etc/config/cuda.amazon.properties [hunk 1]
```diff
@@ -228,6 +228,12 @@ compiler.nvcc123u1.options=--compiler-bindir /opt/compiler-explorer/gcc-10.2.0/b
 compiler.nvcc123u1.nvdisasm=/opt/compiler-explorer/cuda/12.3.1/bin/nvdisasm
 compiler.nvcc123u1.demangler=/opt/compiler-explorer/gcc-10.2.0/bin/c++filt
 compiler.nvcc123u1.objdumper=/opt/compiler-explorer/gcc-10.2.0/bin/objdump
+compiler.nvcc124u1.semver=12.4.1
```
Comment (partouf) [inline etc/config/cuda.amazon.properties:231]: `nvcc124u1` needs to be added to line 12 of this file

Comment (jmuddnv) [reply etc/config/cuda.amazon.properties:231]: So what's the best way to resolve this?  Delete the existing PR and submit a new one?

Comment (partouf) [reply etc/config/cuda.amazon.properties:231]: no need, you can just edit the file in github and push it that way

**Comment by partouf** [comment]
fixed in https://github.com/compiler-explorer/infra/commit/62bfb42450e04948429e6fbd8c5aa4d34a8152ca

**Comment by partouf** [comment]
this is now lvie

**Pull request merged**